src/shared/tpm2-util.c

```diff
@@ -3763,6 +3812,8 @@ int tpm2_create_blob(
         assert(ret_blob_size);
 
         size_t max_size = sizeof(*private) + sizeof(*public);
+        if (seed)
+                max_size += sizeof(*seed);
```

Comment:
maybe make all three parts optional then? seems weird only allowing one of the three to be optional

Reply:
that won't work. this 'blob' is not a standard format, so if all 3 parts were optional there would be no sane way to tell what each part was. Additionally, since this is supposed to store the data for a sealed object, the public and private parts can't be optional - those are always required for a sealed object.

Reply:
indeed